src/mqtt/mqtt_codec.c

```diff
@@ -1802,3 +1923,363 @@ mqtt_msg_dump(mqtt_msg *msg, mqtt_buf *buf, mqtt_buf *packet, bool print_bytes)
 	}
 	return 0;
 }
+
+property *
+property_alloc(void)
+{
+	return nng_zalloc(sizeof(property));
+}
+
+property *
+property_set(struct pos_buf *buf, property *prop, uint8_t prop_id,
```

Comment:
Maybe buf is NULL, check it for safety

Reply:
in most cases, we are editing an existing msg, add property field to a previously allocated msg. please support this usage

Reply:
property_set() just work for decode_properties(), I still need to write a new function to add property and fields.